3rdparty/apg/CMakeLists.txt

```diff
@@ -0,0 +1,5 @@
+add_library(apg_bmp STATIC ${CMAKE_CURRENT_SOURCE_DIR}/apg_bmp.h ${CMAKE_CURRENT_SOURCE_DIR}/apg_bmp.c)
```

Comment:
Will this work when building as a shared library?
I think `CMAKE_POSITION_INDEPENDENT_CODE` should be set in the root cmake script when building as a shared library.